crucible-llvm/src/Lang/Crucible/LLVM.hs

```diff
@@ -53,8 +54,9 @@ llvmGlobals
 llvmGlobals ctx mem = emptyGlobals & insertGlobal var mem
   where var = llvmMemVar $ ctx
 
-llvmExtensionImpl :: HasPtrWidth (ArchWidth arch) => ExtensionImpl p sym (LLVM arch)
-llvmExtensionImpl =
+llvmExtensionImpl :: (HasPtrWidth (ArchWidth arch)) => MemOptions -> ExtensionImpl p sym (LLVM arch)
+llvmExtensionImpl mo =
+  let ?memOpts = mo in
```

Comment:
I'm assuming that this is an implicit param to avoid a much more intrusive implementation that would have threaded this explicitly through lots of places to make it available where it was needed?  If so, you might want to mention that in the PR description for posterity.

Reply:
Yeah... at the moment it doesn't make much difference, but if similar options become more prevalent, I expect it will be a win.